content/en/docs/Installation/trustedboot.md

```diff
@@ -0,0 +1,231 @@
+
+The steps below will guide you into generating the installable assets, and how to re-generate the assets to upgrade the node to a new version of the OS.
+
+### Requirements
+
+- Docker
+- Git
+- A Linux machine with KVM (for testing the images locally)
+
+### Build the container image used to generate keys and installable medium
+
+```bash
+# Build the container image that will be used to generate the keys and installable medium
+git clone https://github.com/kairos-io/enki.git
+cd enki
+docker build -t enki --target tools-image .
+```
+
+### Key generation
+
+To generate the Secure boot certificates and keys run the following commands:
+
+```bash
+# Generate the keys
+docker run -v $PWD/keys:/work/keys -ti --rm enki genkey "$MY_ORG" -o /work/keys
+```
+
+{{% alert title="Warning" %}}
+It is very important to preserve the keys generated in this process in a safe place. Loosing the keys will prevent you to generate new images that can be used for upgrades.
+{{% /alert %}}
+
+### Enroll the keys in Secure Boot
+
+If your machine is in UEFI setup mode Secure Boot keys will be automatically enrolled. 
+
+If UEFI setup mode is not available, you need to enroll the keys manually in the BIOS/UEFI.
+
+This process can vary depending on the vendor, but in general you need to enter the BIOS/UEFI setup during early boot and import the keys, for an example outline you can check the steps for [HPE Hardware](https://techlibrary.hpe.com/docs/iss/proliant-gen10-uefi/GUID-E4427875-D123-4BBF-9056-342168478A02.html).
+
+A video of the process in QEMU is available [here](https://github.com/kairos-io/kairos/assets/2420543/e45f6a08-ec74-4cfd-bdf0-aeb7b23ac9bc).
+
+### Building installable medium
+
+To build the installable medium you need to run the following command:
+
+{{% alert title="Warning" %}}
+
+This method is still a work in progress.
+For now build the testing Kairos iso with:
+
+```bash
+# clone the repo
+git clone https://github.com/kairos-io/kairos
+
+# cd into the repo
+cd kairos
+
+IMAGE=quay.io/kairos/fedora:38-core-amd64-generic-v3.0.0-alpha1
+
+# build the iso with Earthly
+earthly +uki-iso --BASE_IMAGE=$IMAGE
+
+# resulting ISO is in: build/kairos-fedora-38-core-amd64-generic-v3.0.0-alpha1.uki.iso
+
+# Multiple boot options
+earthly +uki-iso --BASE_IMAGE=$IMAGE --ENKI_FLAGS="--cmdline rd.blacklist=i915"
+```
+
+{{% /alert %}}
+
+```bash
+CONTAINER_IMAGE=quay.io/kairos/fedora:38-core-amd64-generic-v3.0.0-alpha1
+docker run --rm -v $PWD/build:/result -v $PWD/keys/:/keys enki build-uki $CONTAINER_IMAGE -o /result/trustedboot.iso -k /keys
+```
+
+### Installation
+
+The installation process is performed as usual and the [Installation instructions]({{< relref "../installation" >}}) can be followed, however the difference is that user-data will be automatically encrypted (both the OEM and the persistent partition) by using the TPM chip and the Trusted Boot mechanism.
```

Comment:
Is there a way to have a remote KMS in the UKI case? Should we mention that here?

Reply:
I would not add it until we tackle it in https://github.com/kairos-io/kairos/issues/2166